src/main/java/org/sejda/sambox/cos/COSFloat.java

```diff
@@ -16,38 +16,32 @@
  */
 package org.sejda.sambox.cos;
 
+import static org.sejda.commons.util.RequireUtils.requireIOCondition;
+
 import java.io.IOException;
 import java.math.BigDecimal;
 import java.util.regex.Pattern;
 
-import static org.sejda.commons.util.RequireUtils.requireIOCondition;
-
 /**
  * This class represents a floating point number in a PDF document.
  *
  * @author Ben Litchfield
  * 
  */
-// @formatter:off
 public class COSFloat extends COSNumber
 {
-    private BigDecimal value;
+    private float value;
```

Comment:
If I'm not mistaken PDFBox also stores the original String representation, since using floats internally will lead to precision loss, and we don't want to output this precision loss, but the original string we read.

So in other words we're fast using floats internally, but with the outside world we need more precision, or at least we can't read 0.3 and then output 0.29983

Reply:
Yes, indeed it was a mistake, I overlooked that behavior in PDFBox. I added a commit that resemble the PDFBox implementation storing the string value read from the file when the conversion is fine (which should be most of the time) and calculates the string representation when the read value needs fixing.